components/features/dashboard/settings/DeleteSchool.tsx

```diff
@@ -0,0 +1,158 @@
+"use client";
+
+import { GraduationCap, LibraryBig, School, Users } from "lucide-react";
+import { useToast } from "@/hooks/use-toast";
+import {
+  Dialog,
+  DialogContent,
+  DialogTrigger,
+  DialogHeader,
+  DialogTitle,
+  DialogDescription,
+  DialogFooter,
+} from "@/components/ui/dialog";
+import axiosInstance from "@/lib/axiosInstance";
+import { getCookie } from "cookies-next";
+import { Button } from "@/components/ui/button";
+import { FormEvent, useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
+import { Input } from "@/components/ui/input";
+import { LoaderCircle } from "lucide-react";
+import { SchoolType } from "@/types/School";
+import { formatNumber } from "@/lib/utils";
+
+type Props = {
+  school: SchoolType;
+};
+
+export default function DeleteSchool({ school }: Props) {
+  const router = useRouter();
+  const { toast } = useToast();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [confirmInputValue, setConfirmInputValue] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [isMatch, setIsMatch] = useState<boolean | undefined>(undefined);
+  const [hasSubmitted, setHasSubmitted] = useState(false);
+  const rule = "Delete " + school.name.trim();
```

Comment:
All school names gets trimmed on creation, so is this necessary?

Reply:
if we remove this it will cause a bug when user edits the school name and then delete the school, the confirmation won't match, but we can prevent this by adding trim on edit school name also. witch approach do you recommend
```suggestion
 onChange={(e) => field.onChange(e.target.value.trim())}
```

Reply:
But isn't schools names are already trimmed on creation and edit?